api/src/main/java/jakarta/data/repository/Pageable.java

```diff
@@ -230,4 +256,88 @@ static Pageable ofSize(int maxPageSize) {
      * @return a new instance of <code>Pageable</code>. This method never returns <code>null</code>.
      */
     Pageable sortBy(Sort... sorts);
+
+    /**
+     * The type of pagination, which can be offset pagination or
+     * keyset cursor pagination which includes a direction.
+     */
+    enum Mode {
```

Comment:
:question: why not move to be a class?
Just curious, because use inner-class here

Reply:
> why not move to be a class?
Just curious, because use inner-class here

Basic common usage of keyset pagination will not require the user to directly interact with `Mode` and `Cursor`, so it's nice to keep these more advanced concepts less prominent as inner classes where they won't overwhelm the user when browsing through the list of API.  A typical user will only need to do:

```
Pageable p1 = Pageable.ofSize(50).sortBy(Sort.asc("id"));
KeysetAwarePage<Product> page1 = products.findBySomething(something, p1);
...
Pageable p2 = p1.afterKeyset(productId); // or page1.nextPageable()
KeysetAwarePage<Product> page2 = products.findBySomething(something, p2);
...
```